sql/mysqld.cc

```diff
@@ -4421,6 +4424,83 @@ a file name for --log-bin-index option", opt_binlog_index_name);
     unireg_abort(MYSQLD_ABORT_EXIT);
   }
 
+  if (opt_libcoredumper)
+  {
+#ifndef HAVE_LIBCOREDUMPER
+    sql_print_warning(
+        "This version of MySQL has not been compiled with "
+        "libcoredumper support, ignoring --coredumper argument");
+#else
+    if (opt_corefile)
+    {
+      sql_print_warning(
+          "Started with --core-file and --coredumper. "
+          "--coredumper will take precedence.");
+    }
+    if (opt_libcoredumper_path != NULL)
+    {
+      /* validate path */
+      if (!is_valid_log_name(opt_libcoredumper_path,
+                             strlen(opt_libcoredumper_path)))
+      {  //filename contain .cnf or .ini on it
+        sql_print_error("Variable --coredumper cannot be set to value %s",
+                        opt_libcoredumper_path);
+        unireg_abort(MYSQLD_ABORT_EXIT);
+      }
+      char   libcoredumper_dir[FN_REFLEN];
+      size_t libcoredumper_dir_length;
+      size_t opt_libcoredumper_path_length= strlen(opt_libcoredumper_path);
+      (void)dirname_part(libcoredumper_dir, opt_libcoredumper_path,
```

Comment:
This works based on the filename, not based on what's actually on the filesystem. I'm not sure if this will be correct if the user specifies a directory without a "/" at the end - I tihink we'll create core files outside of that directory with this logic?

Reply:
This feature tries to mimic binlog, you can ommit the path, pass a directory or a full path with file name. Here if for example the user sets `--coredumper=/var/lib/mysql` mysql will try to write a file on `var/lib` as the last part is recognized as file name(in this case it will be `/var/lib/mysql.yyyymmddhhmmss` ), however if they specify `--coredumper=/var/lib/mysql/` it will write a file under `/var/lib/mysql/` in this case,  `/var/lib/mysql/core.yyyymmddhhmmss`